kasa/protocolfactory.py

```diff
@@ -0,0 +1,39 @@
+"""Module for protocol factory class."""
```

Comment:
I would really much like to avoid this extra file and instead of inline imports if that makes it possible to avoid the circular dependency. Do you think that would be feasible?

Reply:
I've managed to move `get_protocol` inside the `device_factory` module now that it doesn't do any discovery so should be all good here now.